### DISCUSSION TITLE
Add VRF

### DISCUSSION THREAD

**Discussion by samisbakedham** [0xPolygon/polygon-edge#704] [General]
### Description

You will be able to create transactions onchain (msg.vrf.createRandom) that will create a VRF proof and random number amongst other things, VRF proofs/numbers can be then be pulled from any public endpoint for all VRFs generated, specific user generated VRFs etc.

**Comment by laviniat1996** [comment]
Hi @samisbakedham ,

Thank you for this feature request. We apreciate your interest.
This is something we would definitely like to explore in the future. Can you please share with us your current business use-case of VRFs, so we can better understand the need for this feature?

**Reply by samisbakedham** [reply]
Hi, yes. We can use VRF for random number generation etc, instead of relying on Chainlink.
